### PR TITLE
Added new columns in alerts table

### PR DESCRIPTION
**Pull request by tomer-shvadron**
### Description
* Small UI changes
* Added 2 new columns to the Alerts Table - `Subject Type` and `Subject ID`

___

## **Type**
enhancement

___

## **Description**
- Extended the `subject` schema in alerts to include additional fields like `id`, `correlationId`, and `type`.
- Added new columns `Subject Type` and `Subject ID` to the Alerts Table with appropriate rendering logic.
- Updated CSS for better alignment and text handling in Alerts Table.
- Enhanced data structure in alert controller to include new subject fields.
- Updated type definitions to align with new data structure changes.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>TextWithNAFallback.tsx</strong><dd><code>Add Display Name to TextWithNAFallback Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/common/components/atoms/TextWithNAFallback/TextWithNAFallback.tsx
- Added a display name for the `TextWithNAFallback` component.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2334/files#diff-30ac0da6d90ebc0feeca4ca115fb192297a0204279410e21e30044cfdf048c5c">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>fetchers.ts</strong><dd><code>Extend Subject Schema in Alerts List</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/domains/alerts/fetchers.ts
<li>Extended the <code>subject</code> schema to include <code>id</code>, <code>correlationId</code>, and <code>type</code> <br>with specific enum values.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2334/files#diff-3944b935a293fd0baf3d8a9b900c744609f19bedf0187dad6d916658a60c0190">+6/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>AlertsTable.tsx</strong><dd><code>UI Adjustments in Alerts Table</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/AlertsTable.tsx
<li>Adjusted CSS classes for table headers and button elements to improve <br>UI consistency.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2334/files#diff-e6718ae49b4d5f53eb2cc23b8daec84eb2ef5507344147a18eabffd55553d8ca">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>columns.tsx</strong><dd><code>Add New Columns to Alerts Table</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx
<li>Added new columns for <code>Subject Type</code> and <code>Subject ID</code> with respective <br>rendering logic.<br> <li> Utilized <code>useEllipsesWithTitle</code> hook for <code>Subject ID</code> to manage text <br>overflow.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2334/files#diff-8a14ec4f45b577b2033db48193ae99fa39416b35403bd796a1aad5b8458c927c">+36/-1</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>alert.controller.external.ts</strong><dd><code>Enhance Subject Data Structure in Alert Controller</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/alert/alert.controller.external.ts
<li>Included <code>type</code> and <code>correlationId</code> in the subject data structure for <br>business and counterparty.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2334/files#diff-630c3415541ee2aee573e827b8d13f1d9b51e5e0274f68099f4c56d9f69b86bd">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>types.ts</strong><dd><code>Update Type Definitions for Alert Response</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/src/alert/types.ts
<li>Updated type definitions to include <code>correlationId</code> for business and end <br>user.<br>

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2334/files#diff-9ab4f2f23ba35c39a04a2d53416a12cf6fa1bfba20b83cb0e342756302e1edee">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>data-migrations</strong><dd><code>Update Data Migration Commit Reference</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

services/workflows-service/prisma/data-migrations
- Updated subproject commit reference for data migrations.

</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2334/files#diff-9c33eef7eba3ba7ca6006881973adb35d0cbc4048bb7256f945ab6da6f013edf">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

## Summary by CodeRabbit

- **Refactor**
  - Simplified error handling logic by refactoring the `fromClassValidator` method in the `ValidationError` class.
- **Bug Fixes**
  - Replaced `IsDate` import with `IsDateString` in the `transaction-create.dto.ts` file to address date validation issues.
- **Chores**
  - Removed unnecessary imports for `ReviewStatus` and `TransactionRecordStatus` in the `transaction.controller.external.intg.test.ts` file.
- **Documentation**
  - No changes.
- **New Features**
  - Added an export statement for the `InlineRule` type in the `types.ts` file within the `workflows-service`.
- **Style**
  - No changes.
- **Tests**
  - No changes.
- **Revert**
  - No changes.

### CODE REVIEW

File: apps/backoffice-v2/src/domains/alerts/fetchers.ts [hunk 1]
```diff
@@ -76,7 +76,12 @@ export const AlertsListSchema = z.array(
   ObjectWithIdSchema.extend({
     dataTimestamp: z.string().datetime(),
     updatedAt: z.string().datetime(),
-    subject: ObjectWithIdSchema.extend({ name: z.string() }),
+    subject: ObjectWithIdSchema.extend({
+      id: z.string(),
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/domains/alerts/fetchers.ts]: `id` from `ObjectWithIdSchema` is the same validation as far as I can tell. No need for both.
```suggestion

```

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx [hunk 2]
```diff
@@ -84,6 +86,39 @@ export const columns = [
...
+      return (
+        <TextWithNAFallback className={`max-w-[12ch]`}>{titleCase(subjectType)}</TextWithNAFallback>
+      );
+    },
+    header: 'Subject Type',
+  }),
+  columnHelper.accessor('subject.correlationId', {
+    cell: info => {
+      // eslint-disable-next-line react-hooks/rules-of-hooks -- ESLint doesn't like `cell` not being `Cell`.
+      const { ref, styles } = useEllipsesWithTitle<HTMLSpanElement>();
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx:102]: Consider using the `truncate` CSS class instead.

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx [hunk 3]
```diff
@@ -84,6 +86,39 @@ export const columns = [
...
+              variant: 'link',
+              className: '!block cursor-pointer !p-0 text-sm !text-blue-500',
+            })}
+            style={styles}
+            ref={ref}
+          >
+            {subjectId}
+          </TextWithNAFallback>
+        </div>
+      );
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx:118]: Should we have the full id in a tooltip or in the `title` attribute?

File: services/workflows-service/src/alert/alert.controller.external.ts [hunk 4]
```diff
@@ -107,11 +107,15 @@ export class AlertControllerExternal {
         alertDetails: alertDefinition.description,
         subject: counterparty.business
           ? {
+              type: 'business',
               id: counterparty.business.id,
               name: counterparty.business.companyName,
+              correlationId: counterparty.business.correlationId,
             }
           : {
+              type: 'counterparty',
```
Comment (Omri-Levy) [inline services/workflows-service/src/alert/alert.controller.external.ts:116]: A counterparty can be a business, individual, or something else, but the types are just `business` and `counterparty`?

**Review by Omri-Levy — APPROVED** [review]

File: services/workflows-service/src/alert/alert.controller.external.ts (hunk 4 above)
Comment (tomer-shvadron) [reply services/workflows-service/src/alert/alert.controller.external.ts:116]: That's what I understood from Nitzan.

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx (hunk 3 above)
Comment (tomer-shvadron) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx:118]: Already has it.
image

File: apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx (hunk 2 above)
Comment (tomer-shvadron) [reply apps/backoffice-v2/src/pages/TransactionMonitoringAlerts/components/AlertsTable/columns.tsx:102]: This is actually what makes the tooltip exist so I will leave it for now. We can review in the future.

**Pull request merged**